src/key-value.spec.js

```diff
@@ -0,0 +1,23 @@
+describe('ReactKeyValue', () => {
+  it('should have the proper class name');
+  it('should set the default props');
+  it('should get the initial state from the props');
+  it('should handle adding new rows');
+  it('should handle when a key changes');
+  it('should handle when a value changes');
+  it('should handle removing a row');
+  it('should render a KeyItem');
+  it('should render a ValueItem');
+  it('should have a renderRows method');
+  it('should have a renderAddButton method');
```

Comment:
it doesn't matter what is responsible for rendering an "add new button". The point is to render an "add new button". How you achieve it is just an implementation detail which shouldn't be tested, just the result. I'd also test what happens when `props. customAddButtonRenderer ` is set.